pkg/ctl/topic/set_retention.go

```diff
@@ -0,0 +1,197 @@
+)
+
+func SetRetentionCmd(vc *cmdutils.VerbCmd) {
+	desc := cmdutils.LongDescription{}
+	desc.CommandUsedFor = "Set the retention policy for a topic"
+	desc.CommandPermission = "This command requires tenant admin permissions."
+
+	var examples []cmdutils.Example
+	removeRetention := cmdutils.Example{
+		Desc:    "Set the retention policy for a topic",
+		Command: "pulsarctl topics set-retention tenant/namespace/topic",
+	}
+	examples = append(examples, removeRetention)
+	desc.CommandExamples = examples
+
+	var out []cmdutils.Output
+	successOut := cmdutils.Output{
+		Desc: "normal output",
+		Out:  "Set the retention policy for [topic] successfully",
+	}
+
+	noTopicName := cmdutils.Output{
+		Desc: "you must specify a tenant/namespace/topic name, please check if the tenant/namespace/topic name is provided",
+		Out:  "[✖]  the topic name is not specified or the topic name is specified more than one",
+	}
+
+	tenantNotExistError := cmdutils.Output{
+		Desc: "the tenant does not exist",
+		Out:  "[✖]  code: 404 reason: Tenant does not exist",
+	}
+
+	nsNotExistError := cmdutils.Output{
+		Desc: "the namespace does not exist",
+		Out:  "[✖]  code: 404 reason: Namespace (tenant/namespace) does not exist",
+	}
+
+	out = append(out, successOut, noTopicName, tenantNotExistError, nsNotExistError)
+	desc.CommandOutput = out
+
+	vc.SetDescription(
+		"set-retention",
+		"Set the retention policy for a topic",
+		desc.ToString(),
+		desc.ExampleToString(),
+		"set-retention",
+	)
+
+	var timeStr string
+	var sizeStr string
+
+	vc.Command.Flags().StringVarP(&timeStr, "time", "", "",
+		"Retention time in minutes (or minutes, hours, days, weeks eg: 100m, 3h, 2d, 5w). "+
+			"0 means no retention and -1 means infinite time retention")
+	vc.Command.Flags().StringVarP(&sizeStr, "size", "", "",
+		"Retention size limit (eg: 10M, 16G, 3T). "+
+			"0 or less than 1MB means no retention and -1 means infinite size retention")
+
+	vc.SetRunFuncWithNameArg(func() error {
+		return doSetRetention(vc, timeStr, sizeStr)
+	}, "the topic name is not specified or the topic name is specified more than one")
+}
+
+func doSetRetention(vc *cmdutils.VerbCmd, timeStr string, sizeStr string) error {
+	topic := vc.NameArg
+
+	if timeStr == "" {
+		return errors.New("missing option: --time")
+	}
+	if sizeStr == "" {
+		return errors.New("missing option: --size")
+	}
+
+	duration, err := parseRetentionTimeString(timeStr)
+	if err != nil {
+		return err
+	}
+
+	size, err := parseRetentionSizeString(sizeStr)
+	if err != nil {
+		return err
+	}
+
+	admin := cmdutils.NewPulsarClient()
+	topicName, err := util.GetTopicName(topic)
+	if err != nil {
+		return err
+	}
+
+	err = admin.Topics().SetRetention(*topicName, util.RetentionData{
+		RetentionSizeInMB:      size,
+		RetentionTimeInMinutes: duration,
+	})
+	if err == nil {
+		vc.Command.Printf("Set the retention policy successfully on [%s]\n", topic)
+	}
+
+	return err
+}
+
+func parseRetentionSizeString(size string) (int64, error) {
```

Comment:
I think we already have this method in utils package, why not use that?

Reply:
I don't found this method, could you point this method?